packages/common/src/search/_internal/hubSearchPosts.ts

```diff
@@ -0,0 +1,99 @@
+ * Currently supported filters include (filter predicate key => accepted values):
+ *   - access: 'public' | 'private' | 'org' | Array<'public' | 'private' | 'org'>
+ *   - body: string
+ *   - channel: string | string[] (mapped internally to `channels`)
+ *   - owner: string | string[] (first value mapped to `creator`)
+ *   - discussion: string | string[]
+ *   - editor: string | string[]
+ *   - status: 'pending' | 'approved' | 'rejected' | 'deleted' | 'hidden' | 'blocked' | Array<...>
+ *   - title: string | string[]
+ *   - parentId: string | string[] (mapped internally to `parents`)
+ *   - groups: string | string[]
+ *   - postType: 'text' | 'announcement' | 'poll' | 'question'
+ *   - created: IDateRange<string | number> (mapped to createdAfter / createdBefore)
+ *   - modified: IDateRange<string | number> (mapped to updatedAfter / updatedBefore)
+ *
+ * Notes / constraints:
+ *   - For single-value filters where an array is provided (e.g., owner, editor, discussion, title, body), only the first value is used.
+ *   - Date range predicates use the shape: { from: string|number; to: string|number }.
+ *   - Unsupported filters are ignored silently.
+ *
+ * INCLUDE SUPPORT (options.include tokens mapped to post relations - bare tokens only)
+ *   channel     -> PostRelation.CHANNEL
+ *   parent      -> PostRelation.PARENT
+ *   replies     -> PostRelation.REPLIES (returns replies collection or paging info)
+ *   replyCount  -> PostRelation.REPLY_COUNT (numeric aggregate)
+ *   reactions   -> PostRelation.REACTIONS
+ *   channelAcl  -> PostRelation.CHANNEL_ACL
+ *   Unknown tokens are ignored; no namespaced forms (e.g. `post:channel`) are supported.
+ *
+ * Currently supported sort fields include (pass via `options.sortField`):
+ *   - body
+ *   - channelId
+ *   - created  (maps to createdAt)
+ *   - modified (maps to updatedAt)
+ *   - owner    (maps to creator)
+ *   - discussion
+ *   - editor
+ *   - id
+ *   - parentId
+ *   - status
+ *   - title
+ * Sort order: 'ASC' | 'DESC' (via `options.sortOrder`).
+ * Pagination options: `options.num` (page size), `options.start` (1-based offset handled by API).
+ *
+ * @param query An IQuery object (must include `filters` and `targetEntity: 'post'` upstream)
+ * @param options An IHubSearchOptions object
+ * @returns a promise that resolves an IHubSearchResponse<IHubSearchResult>
+ */
+export async function hubSearchPosts(
+  query: IQuery,
+  options: IHubSearchOptions
+): Promise<IHubSearchResponse<IHubSearchResult>> {
+  const processedFilters = processPostFilters(query.filters);
+  const processedOptions = processPostOptions(options);
+  const processedRelations = processRelations(options.include);
+
+  const data: ISearchPosts = {
+    ...processedFilters,
+    ...processedOptions,
+    relations: processedRelations,
+  };
+  const { items, nextStart, total } = await searchPostsV2({
+    ...options.requestOptions,
+    token: options.requestOptions?.hubApiKey,
```

Comment:
The `hubApiKey` is passed here to `searchPostsV2`. The token is defined as a prop in `IDiscussionsRequestOptions`, which `ISearchPostsParams` extends.